### PR TITLE
Duplo-20877 RDS Performance Insight

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added support to enable Performance Insights to duplocloud_rds_instance
## Summary of changes

This PR does the following:

- Added performance_insights schema attributes
- Implemented API request for performance_insights
- Updated documentation

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

___

### **PR Type**
enhancement, documentation

___

### **Description**
- Added support for Amazon RDS Performance Insights in the `duplocloud_rds_instance` resource.
- Introduced new schema attributes for configuring Performance Insights, including enabling/disabling, KMS key ID, and retention period.
- Updated the RDS instance creation and update logic to handle Performance Insights settings.
- Enhanced documentation to include details and examples of the new Performance Insights feature.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_rds_instance.go</strong><dd><code>Add RDS Performance Insights support in resource schema</code>&nbsp; &nbsp; </dd></summary>
<hr>

duplocloud/resource_duplo_rds_instance.go

<li>Added support for RDS Performance Insights configuration.<br> <li> Implemented functions to handle Performance Insights settings.<br> <li> Updated RDS instance creation and update logic to include Performance <br>Insights.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/650/files#diff-a8bf9b56a3e9b39899a5d37d7608110678908754e0848fab2d0320653ea65f71">+124/-9</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rds_instance.go</strong><dd><code>Implement Performance Insights API support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

duplosdk/rds_instance.go

<li>Added new structs for Performance Insights configuration.<br> <li> Implemented API update methods for Performance Insights.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/650/files#diff-3aeb597ae11ae40342366a786a3b78f55a1265ea03d8d2729401d9a7f3b7291a">+74/-36</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>rds_instance.md</strong><dd><code>Update documentation for RDS Performance Insights</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/resources/rds_instance.md

<li>Documented new <code>performance_insights</code> attribute.<br> <li> Added example usage for Performance Insights.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/650/files#diff-2be785952795586c48f30db78e5f3684f72a430f66155dd8ea0d85f0607eedb7">+33/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource.tf</strong><dd><code>Add example for Performance Insights configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/resources/duplocloud_rds_instance/resource.tf

- Added example configuration for RDS Performance Insights.

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/650/files#diff-4461dc184c959908eb1db51855cf42bfafd849c340d3632e8da923f5a5a6240d">+22/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by pravin-duplocloud — APPROVED** [review]
LGTM

File: duplocloud/resource_duplo_rds_instance.go [hunk 1]
```diff
@@ -289,6 +289,38 @@ func rdsInstanceSchema() map[string]*schema.Schema {
...
 		},
+		"performance_insights": {
+			Description: "Amazon RDS Performance Insights is a database performance tuning and monitoring feature that helps you quickly assess the load on your database, and determine when and where to take action. Perfomance Insights get apply when enable is set to true. Not applicable for Cluster Db",
+			Type:        schema.TypeList,
+			MaxItems:    1,
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"enable": {
+						Description: "Enable or Disable Performance Insighta",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_rds_instance.go:300]: Typo `Insighta`

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_rds_instance.go]: done

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**